cmd/cli/main.go

```diff
@@ -0,0 +1,375 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	v1 "github.com/k8up-io/k8up/v2/api/v1"
+	"github.com/k8up-io/k8up/v2/cli/restore"
+	"github.com/k8up-io/k8up/v2/cmd"
+	"github.com/urfave/cli/v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	Command = &cli.Command{
+		Name:        "cli",
+		Description: "CLI commands that can be executed everywhere, currently just restore is supported",
+		Subcommands: []*cli.Command{
+			{
+				Name:   "restore",
+				Action: RunRestore,
+				Flags: []cli.Flag{
+					&cli.StringFlag{
+						Destination: &restore.Cfg.Snapshot,
+						Required:    false,
+						Name:        "snapshot",
+						Value:       "latest",
+						Usage:       "Optional ; ID of the snapshot `kubectl get snapshots`, if left empty 'latest' will be used, set itvia cli or via env: ",
+						EnvVars: []string{
+							"SNAPSHOT",
+						},
+					},
```

Comment:
Can we somehow reduce the boilerplate code by adding a function with global vars which will create the object for us? It will be more succinct, less boilerplate and better for readability.

Reply:
explain please

Reply:
Suggestion `getFlag(type, globVar1, globVar2....) `or `getStringFlag(globVar1, globVar2....)`. Try this one and see if it helps with readability and with reduced boilerplate code.

Reply:
it's yet another place where something can fail, I don't agree it'll solve any problem, but only introduce new, this code must be somewhere and adding another layer of abstraction won't solve any issue

Reply:
I wouldn't add something like this here. All other commands in K8up use this pattern. Using a new pattern here breaks consistency.

We could reduce the amount of lines by making all the structs one-liner like in `restic/main.go` but I actually think this here is more readable.

Reply:
Then lets keep it as is.